crates/api_crud/src/multi_community/create.rs

```diff
@@ -0,0 +1,40 @@
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
+use lemmy_api_common::{community::CreateMultiCommunity, context::LemmyContext, utils::slur_regex};
+use lemmy_db_schema::{
+  source::multi_community::{MultiCommunity, MultiCommunityInsertForm},
+  traits::Crud,
+};
+use lemmy_db_views_local_user::LocalUserView;
+use lemmy_db_views_site::SiteView;
+use lemmy_utils::{
+  error::LemmyResult,
+  utils::{slurs::check_slurs, validation::is_valid_display_name},
+};
+use url::Url;
+
+pub async fn create_multi_community(
+  data: Json<CreateMultiCommunity>,
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<MultiCommunity>> {
+  let local_site = SiteView::read_local(&mut context.pool()).await?.local_site;
+  is_valid_display_name(&data.name, local_site.actor_name_max_length)?;
+
+  let slur_regex = slur_regex(&context).await?;
+  check_slurs(&data.name, &slur_regex)?;
+
+  let form = MultiCommunityInsertForm {
+    creator_id: local_user_view.person.id,
+    name: data.name.clone(),
+    title: data.title.clone(),
+    description: data.description.clone(),
+    ap_id: Url::parse(&format!(
+      "{}/m/{}",
+      &local_user_view.person.ap_id, &data.name
+    ))?
+    .into(),
```

Comment:
Don't we have some trait / helper functions to create these somewhere?

Okay I see that both `Community` and `Person` impls have a `local_url` function. It'd be a good idea to make this function a trait, and have community, person, and multi_community implement it.

Reply:
Those are called from multiple places, but here its only used in a single place. Would be unnecessary to add a separate method.